rskj-core/src/main/java/co/rsk/net/BlockSyncService.java

```diff
@@ -69,21 +70,19 @@ public BlockSyncService(
         this.config = config;
     }
 
-    public BlockProcessResult processBlock(@Nonnull Block block, Peer sender, boolean ignoreMissingHashes) {
-        Instant start = Instant.now();
-        long bestBlockNumber = this.getBestBlockNumber();
-        long blockNumber = block.getNumber();
+    protected boolean preprocessBlock(@Nonnull Block block, Peer sender, boolean ignoreMissingHashes) {
```

Comment:
Why not to check if the block is ALREADY in the "net store"? Maybe it could avoid the process of duplicated blocks that arrives from different peers to this node, but are not yet added to any chain because they are still in the to process queue

Reply:
I'm also worried about this.

Receiving the same block multiple times doesn't seem correctly handled.

Reply:
Done